build.assets/charts/Dockerfile-distroless

```diff
@@ -6,11 +6,21 @@ COPY fetch-debs ./
 RUN ./fetch-debs dumb-init libpam0g libaudit1 libcap-ng0
 
 FROM debian:11 AS teleport
-# NOTE that the TELEPORT_RELEASE_INFIX *must* include the leading dash if set
+# Install the teleport binary from an architecture-specific debian package. Note
+# that we cannot simply pass a ready-made package filename in as a build-arg, as
+# this dockerfile is used for a multiarch build and any build-args will be
+# re-used for multiple ${TARGETARCH}es. In order to get around this we pass
+# various components of the architecture-specific debian package filename in as
+# individual build args and re-assemble it *inside* the build.
+
+# NOTE that the TELEPORT_RELEASE_INFIX and TELEPORT_FIPS_INFIX 
+# *must* include the leading dash if set.
 ARG TELEPORT_RELEASE_INFIX
+ARG TELEPORT_FIPS_INFIX=""
 ARG TELEPORT_VERSION
+# TARGETARCH is supplied by the `buildx` mechanics
 ARG TARGETARCH
-ENV TELEPORT_DEB_FILE_NAME=teleport${TELEPORT_RELEASE_INFIX}_${TELEPORT_VERSION}_${TARGETARCH}.deb
+ENV TELEPORT_DEB_FILE_NAME=teleport${TELEPORT_RELEASE_INFIX}_${TELEPORT_VERSION}${TELEPORT_FIPS_INFIX}_${TARGETARCH}.deb
```

Comment:
Do the three args actually need to be separate? Nothing seems to use them on their own, so perhaps just:

    ARG TELEPORT_DEB_BASENAME
    ENV TELEPORT_DEB_FILENAME=${TELEPORT_DEB_BASENAME}_${TARGETARCH).deb

and then pass `teleport-v13.0.0` or `teleport-ent-v13.0.0` or `teleport-ent-v13.0.0-fips`, etc as the arg?

Alternatively, since I expect that release-infix and fips-infix are used for `-ent` and `-fips` respectively and don't need to be anything else, you could do:

```
ARG TELEPORT_VERSION
ARG ENTERPRISE=""
ARG FIPS=""
ENV TELEPORT_DEB_FILE_NAME=teleport${ENTERPRISE:+-ent}_${TELEPORT_VERSION}${FIPS:+-fips}_${TARGETARCH}.deb
```

That's a bit of a mouthful but it does mean you can build with `--build-arg FIPS=1 --build-arg ENTERPRISE=1` for example, which is a bit clearer when running `docker build`.

Reply:
I'm going to run with `TELEPORT_DEB_BASENAME` abns see how it comes out

Reply:
Actually - looking at how this gets called, there is a lot mpre surgery to do than I'd like right now. I ,ight leave it as-is for now but raise a ticket to revisit.